### PR TITLE
Fix bug with empty Salesforce result table

### LINKED ISSUE #583: RPA.Salesforce - Salesforce Query Result As Table doesn´t work properly when no objects fond
I have noticed something wrong in the Salesforce library. The keyword Salesforce Query Result As Table doesn´t work like the keyword Salesforce Query.
Salesforce Query Result As Table fails when there are now rows matching the query.
I am attaching part of my loh.html file in a screenshot. Please, let me know if you want me to open an incident or a case.

image (1)

### PR DESCRIPTION
**Pull request by cmin764**
Closes #583 

### ToDo
- [x] Fix docstrings and provide better understanding on `Salesforce Query    as_table=${True}` vs. `Salesforce Query Result As Table`
- [ ] Add unit-test for empty result case
- [x] Release notes & main version bump (then release)

### CODE REVIEW

**Comment by cmin764** [comment]
Tested just fine with robot [example](https://github.com/cmin764/robots/blob/main/enterprise/salesforce.robot#L21-L30):
```robotframework
Query Empty Result As Table
    ${acc_id} =    Set Variable    '0017Q00000MTmkiQAE'  # '0017Q00000MTmkiQAD'
    ${query} =    Set Variable
    ...    SELECT Name FROM Opportunity WHERE AccountId = ${acc_id}

    ${result} =    Salesforce Query    ${query}    as_table=${True}
    Log To Console    ${result}

    ${result} =    Salesforce Query Result As Table    ${query}
    Log To Console    ${result}
```

File: packages/main/pyproject.toml [hunk 1]
```diff
@@ -57,7 +57,7 @@ pynput-robocorp-fork = "^4.0.0"
 python-xlib = { version = ">=0.17", platform = "linux" }
 psutil = { version = "^5.7.0", platform = "win32" }
 pyperclip = "^1.8.0"
-exchangelib = "^4.5.1"
+exchangelib = "^4.7.5"
```
Comment (cmin764) [inline packages/main/pyproject.toml:60]: Meanwhile, a bug affecting OAuth2 token refresh with Microsoft Exchange library was fixed. (including in this release as opportunistic refactoring)

**Review by bogdancondurache — APPROVED** [review]
